src/gui/plugins/axes/AxesConfig.qml

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+import QtQuick 2.9
+import QtQuick.Controls 2.1
+import QtQuick.Dialogs 1.0
+import QtQuick.Layouts 1.3
+import "qrc:/qml"
+
+GridLayout {
+  columns: 4
+  columnSpacing: 10
+  Layout.minimumWidth: 200
+  Layout.minimumHeight: 500
+  anchors.fill: parent
+  anchors.leftMargin: 10
+  anchors.rightMargin: 10
+
+  // Left spacer
+  Item {
+    Layout.columnSpan: 1
+    Layout.rowSpan: 1
+    Layout.fillWidth: true
+  }
+
+  CheckBox {
+    Layout.alignment: Qt.AlignHCenter
+    Layout.columnSpan: 2
+    text: qsTr("Show/Hide Axes")
+    checked: true
+    onClicked: {
+      AxesConfig.OnShow(checked)
+    }
+  }
+
+  // Right spacer
+  Item {
+    Layout.columnSpan: 1
+    Layout.rowSpan: 1
+    Layout.fillWidth: true
+  }
+
+  // Left spacer
+  Item {
+    Layout.columnSpan: 1
+    Layout.rowSpan: 1
+    Layout.fillWidth: true
+  }
+
+  CheckBox {
+    Layout.alignment: Qt.AlignHCenter
+    Layout.columnSpan: 2
+    text: qsTr("Lines/Arrow Axes")
+    checked: true
+    onClicked: {
+      AxesConfig.OnTypeAxes(checked)
+    }
+  }
+
+  // Right spacer
+  Item {
+    Layout.columnSpan: 1
+    Layout.rowSpan: 1
+    Layout.fillWidth: true
+  }
+
+  // Left spacer
+  Item {
+    Layout.columnSpan: 1
+    Layout.rowSpan: 1
+    Layout.fillWidth: true
+  }
+
+  ComboBox {
+    Layout.alignment: Qt.AlignHCenter
+    Layout.columnSpan: 2
+    objectName: "ComboBoxEntities"
+    model: AxesConfig.comboList
+    onActivated: {
+      AxesConfig.onCurrentIndexChanged(index)
+      axesLength.value = AxesConfig.length
+      x.value = AxesConfig.axesX
+      y.value = AxesConfig.axesY
+      z.value = AxesConfig.axesZ
+      roll.value = AxesConfig.axesRoll
+      pitch.value = AxesConfig.axesPitch
+      yaw.value = AxesConfig.axesYaw
+    }
+    onAccepted: {
```

Comment:
Each time I choose a new model, a new visual is created. Is that intentional? I end up with many of them

![image](https://user-images.githubusercontent.com/5751272/91506995-75562e80-e888-11ea-8c19-6e05b1841cf0.png)

Reply:
@chapulina, yes, this is intentional. The GUI allows you to check or uncheck the `Show Axes` option that enables the axes for the selected entity. I did it like this because you may want to visualize more than one axes on the scene.

Do you expect a different behaviour ?

Reply:
I see. The usage is still a bit not intuitive to me. For example, if multiple models have axes showing, and I hide the axes for one of them. When I change to another entity, that entity's axes disappear:

![axes](https://user-images.githubusercontent.com/5751272/108132951-98f47780-7068-11eb-88c4-fd6344fd1dd2.gif)

I'd expect that when I select the "sphere" from the dropdown, the sphere keeps its axes showing, and the "Show Axes" box checks itself. That is, the widget updates itself to match the currently model on the dropdown.